polyfill/test/duration.mjs

```diff
@@ -139,21 +139,37 @@ describe('Duration', () => {
       equal(`${Duration.from('P1Y1M1W1DT1H1M1.1234567S')}`, 'P1Y1M1W1DT1H1M1.123456700S');
       equal(`${Duration.from('P1Y1M1W1DT1H1M1.12345678S')}`, 'P1Y1M1W1DT1H1M1.123456780S');
       equal(`${Duration.from('P1Y1M1W1DT1H1M1.123456789S')}`, 'P1Y1M1W1DT1H1M1.123456789S');
+      equal(`${Duration.from('P1Y1M1W1DT1H1M1.123456789123S')}`, 'P1Y1M1W1DT1H1M1.123456789S');
     });
     it('variant decimal separator', () => {
       equal(`${Duration.from('P1Y1M1W1DT1H1M1,12S')}`, 'P1Y1M1W1DT1H1M1.120S');
     });
-    it('decimal places only allowed in seconds', () => {
+    it('decimal places only allowed in time units', () => {
       [
         'P0.5Y',
         'P1Y0,5M',
         'P1Y1M0.5W',
         'P1Y1M1W0,5D',
-        'P1Y1M1W1DT0.5H',
-        'P1Y1M1W1DT1H0,5M',
-        'P1Y1M1W1DT1H0.5M0.5S'
+        { years: 0.5 },
```

Comment:
IMHO, all these rounding tests should be done with both positive and negative durations, because a frequent cause of rounding errors in libraries is differences in handling negative vs. positive numbers.